.github/workflows/oaf-check.yml

```diff
@@ -11,18 +11,28 @@ jobs:
   oaf-up-to-date:
     name: Check for new OAF version
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
           python-version: ${{ inputs.python-version }}
           cache: 'pip'
-          cache-dependency-path: 'requirements/dev.txt'
+          cache-dependency-path: 'requirements/*.txt'
       - name: Install dependencies
         run: pip install $(grep "pip-tools==" requirements/dev.txt)
       - name: Run compile dependencies
         run: ./bin/compile_dependencies.sh --upgrade-package open-api-framework
-
       - name: Check git diff
         run: git diff --exit-code -- requirements/*.txt
+      - name: Save OAF Version
+        # save the package version for branch names and commit messages
+        if: failure()
+        id: save-oaf-version
+        run: echo "OAF_VERSION=$(grep -Po '(?<=open-api-framework==)[^\n]*' requirements/dev.txt)" >> $GITHUB_ENV
+      - name: Create Update PR
+        if: failure()
```

Comment:
For my understanding, is this `failure()` needed because you want this to run only if `Check git diff` failed? Because `Save OAF Version` shouldn't return a failure status code, right?

Reply:
Changed